src/Symfony/Bundle/TwigBundle/DependencyInjection/TwigExtension.php

```diff
@@ -132,6 +132,11 @@ public function load(array $configs, ContainerBuilder $container)
 
         $container->getDefinition('twig')->replaceArgument(1, $config);
 
+        if (false === $config['cache']) {
+            $container->removeDefinition('twig.cache_warmer');
+            $container->removeDefinition('twig.template_cache_warmer');
```

Comment:
Does someone remember why we have two cache warmers? Both classes aim to solve the same problem. Is it still necessary to keep both?

Reply:
I looked at this a LONG time ago for #24608, but I can't remember what my conclusion was. Iirc, they DO indeed do the same thing, but maybe one has one slightly additional feature. Not too helpful, I know :)

Reply:
IIRC `TemplateCacheCacheWarmer` is the dedicated warmer for `templating` layer (if enabled), and it's necessary while this layer exists. In the other hand, `TemplateCacheWarmer` is the new version without `templating` layer/dependecy.